client/cody/src/chat/utils.ts

```diff
@@ -81,3 +83,38 @@ export async function getCodebaseContext(
     const embeddingsSearch = repoId && !isError(repoId) ? new SourcegraphEmbeddingsSearchClient(client, repoId) : null
     return new CodebaseContext(config, codebase, embeddingsSearch, new LocalKeywordContextFetcher(rgPath, editor))
 }
+
+export async function getAuthStatus(
+    config: Pick<ConfigurationWithAccessToken, 'serverEndpoint' | 'accessToken' | 'customHeaders'>
+): Promise<AuthStatus> {
+    if (!config.accessToken) {
+        return {
+            showInvalidAccessTokenError: false,
+            authenticated: false,
+            hasVerifiedEmail: false,
+            requiresVerifiedEmail: false,
+            onSupportedSiteVersion: false,
+        }
+    }
+    const client = new SourcegraphGraphQLAPIClient(config)
+    if (client.isDotCom() || isLocalApp(config.serverEndpoint)) {
+        const data = await client.getCurrentUserIdAndVerifiedEmail()
+        return {
+            showInvalidAccessTokenError: isError(data),
+            authenticated: !isError(data),
+            hasVerifiedEmail: !isError(data) && data?.hasVerifiedEmail,
+            // on sourcegraph.com this is always true
+            requiresVerifiedEmail: true,
+            onSupportedSiteVersion: true,
+        }
+    }
+    const currentUserID = await client.getCurrentUserId()
+    const instanceVersion = await client.getSiteVersion()
```

Comment:
Everywhere else you call it "site version" so don't make a distinction here, just call it siteVersion.
```suggestion
    const siteVersion = await client.getSiteVersion()
```